2020/04/28/convenient-reporting-with-jasper.html.md

```diff
@@ -0,0 +1,343 @@
+---
+author: "Árpád Lajos"
+title: "Convenient reporting with Jasper"
+tags: java, reporting, jasper, wildfly
+
+---
+
+Business Intelligence (BI) and reporting is a huge problem space in custom software. There's a wide variety of business needs for looking at past and predictive behavior and building a reporting tool can be a much more cost effective way to get this data, instead of writing individual queries or manually generating reports.  
+
+I've been working with Jasper in the Java project space and wanted to write about some research I've collected on the topic. 
+
+JasperReports uses .jrxml files as input and a report as output. Possible output targets include:
+
+- Screen
+- Printer
+- PDF
+- HTML
+- Excel files
+- RTF
+- ODT
+- CSV
+- XML
+
+##Jasper history
+
+- June 2001: Teodor Danciu began working on JasperReports
+- September 2001: [SourceForge](https://sourceforge.net/) was registered
+- November 3, 2001: JasperReports 0.1.5 was released
+- 2004: Panscopic teamed up with Teodor Danciu, acquired intellectual ownership of the product and changed its name to [JasperSoft](https://www.jaspersoft.com/)
+- 2007: Brian Gentile became CEO of the company
+- July 21, 2005: JasperReports 1.0 was released
```

Comment:
@jonjensen good catch, thanks. The dates are correct, but they are out of order indeed.

Reply:
I think we can just show the year for the 2005 & 2014 events, since there are no other events in those years and none of our readers care what exact date those happened so far in the past.

Reply:
@jonjensen I think you are right. It is desirable to apply some standards and maybe simplicity in this case is helpful for the readers.